executor/src/witgen/mod.rs

```diff
@@ -85,12 +81,8 @@ impl<'a, 'b, T: FieldElement, Q: QueryCallback<T>> WitnessGenerator<'a, 'b, T, Q
     /// Generates the committed polynomial values
     /// @returns the values (in source order) and the degree of the polynomials.
     pub fn generate(self) -> Vec<(&'a str, Vec<T>)> {
-        if self.degree.is_zero() {
-            panic!("Resulting degree is zero. Please ensure that there is at least one non-constant fixed column to set the degree.");
-        }
```

Comment:
Shouldn't we keep this?

Reply:
I don't think so: having degree zero is being used as some sort of error management, but it should not be. Having degree zero here should just return empty columns.